### PR TITLE
Cleanup machine definitions, support Odroid-C2

### PR DESCRIPTION
**Pull request by rbray89**
Would like to cleanup the machine definitions to better tie them to their architectures. 
Would also like to add support for the odroidc2

### CODE REVIEW

**Comment by rbray89** [comment]
@pvizeli Could you please look at this change?

**Comment by pvizeli** [comment]
I'm fine with this. Actual it's xu3/4 in progress. You need also provide a dockerfile for machine

File: builder/builder.sh [hunk 1]
```diff
@@ -23,7 +23,17 @@ IMAGE=""
 BUILD_LIST=()
 BUILD_TYPE="addon"
 BUILD_TASKS=()
-BUILD_MACHINE=('raspberrypi' 'raspberrypi2' 'raspberrypi3' 'raspberrypi3-64' 'intel-nuc' 'qemux86-64' 'qemux86' 'qemuarm' 'qemuarm-64' 'tinker')
+declare -A BUILD_MACHINE=([odroidc2]=aarch64 \
```
Comment (pvizeli) [inline builder/builder.sh]: Do we ned declare this and can not use the same schema as before?

Comment (rbray89) [reply builder/builder.sh]: I did it this way as we'll start adding a lot more machines, and I didn't want to keep adding to the pattern to determine the architecture. This way, they are tied together in a single, easy to read and edit.

**Comment by rbray89** [comment]
@pvizeli Where are the dockerfiles? I can't seem to fine where they reside.

**Comment by pvizeli** [comment]
https://github.com/home-assistant/hassio-homeassistant

**Comment by pvizeli** [comment]
I don't know why, but I don't work :(

**Comment by rbray89** [comment]
Ah, I see you found the issue, good work!

**Comment by pvizeli** [comment]
yes, it should work now

**Pull request merged**